Cargo.toml

```diff
@@ -47,7 +47,9 @@ r2d2 = { version = "0.8.8", optional = true }
 
 # Only needed for cluster
 crc16 = { version = "0.4", optional = true }
+itertools = { version = "0.10", optional = true }
```

Comment:
What are we pulling in itertools for? If it's just `unique()`, I'd prefer some kind of hand-rolled version of that if it isn't too much code (say, less than 100).

Reply:
Done.